src/types/InvertPattern.ts

```diff
@@ -318,7 +318,9 @@ type InvertPatternForExcludeInternal<p, i, empty = never> =
     ? {
         select: InvertPatternForExcludeInternal<subpattern, i, empty>;
         array: i extends readonly (infer ii)[]
-          ? InvertPatternForExcludeInternal<subpattern, ii, empty>[]
+          ? i extends any[]
+            ? InvertPatternForExcludeInternal<subpattern, ii, empty>[]
+            : readonly InvertPatternForExcludeInternal<subpattern, ii, empty>[]
```

Comment:
**Stylistic**: There's a pattern of using `MaybeAddReadonly<T>` in this file that I think we should keep:

```suggestion
          ? MaybeAddReadonly<
              InvertPatternForExcludeInternal<subpattern, ii, empty>[],
              IsReadonlyArray<i>
            >
```

Reply:
@gvergnaud 
it looks good! Included it in this commit [979b358](https://github.com/gvergnaud/ts-pattern/pull/284/commits/979b358fbb5fbe14cc2af9d6da1d09bcfad18830)